src/Transports/MassTransit.SqlTransport.PostgreSql/SqlTransport/PostgreSql/PostgresDbConnectionContext.cs

```diff
@@ -226,10 +226,10 @@ await connection.Connection.ExecuteScalarAsync<int>($"LISTEN transport_msg_{queu
                                     if (queueIds.Contains(queueId))
                                         break;
 
-                                    await connection.Connection.ExecuteScalarAsync<int>($"LISTEN transport_msg_{queueId}", Stopping)
+                                    await connection.Connection.ExecuteScalarAsync<int>($"LISTEN {_context.Schema}_msg_{queueId}", Stopping)
```

Comment:
I think you might need to SQL-ize this as a schema can have `-` in it? And that wouldn't be a legal channel name?

Reply:
There is also a relevant `pg_notify` call in the SQL code that would need to use the same schema name?

Reply:
A function for C# that purifies the schema name and uses that for the channel name when generating the SQL and when called by the C# client to listen would be best. Just swap - for _ as usual, or whatever other invalid characters are in the schema name.

Reply:
After long testing I realized that both schema and channel names are considered identifiers in Postgres and they have the same validation rules: https://www.postgresql.org/docs/current/sql-syntax-lexical.html#SQL-SYNTAX-IDENTIFIERS

So what we wrote above is not correct. 

What I've noticed is that since both schemas and channels have 63 characters limit. In the current MassTransit implementation if a schema name is greater than 39 characters it will break when we append _msg_ queueId. I created a function to truncate the names that are longer and created a test for it. 

Additionally the name of the channel should be enclosed in quotes to avoid any issues.

Please have a look @phatboyg and let me know if any other changes are required